httpclient-backend/src/main/scala/sttp/client/httpclient/HttpClientBackend.scala

```diff
@@ -106,17 +110,22 @@ abstract class HttpClientBackend[F[_], S](client: HttpClient, closeClient: Boole
 
     val encoding = headers.collectFirst { case h if h.is(HeaderNames.ContentEncoding) => h.value }
     val method = Method(res.request().method())
-    val byteBody = if (encoding.contains("gzip") && method != Method.HEAD) {
-      new GZIPInputStream(res.body())
-    } else if (encoding.contains("deflate") && method != Method.HEAD) {
-      new InflaterInputStream(res.body())
+    val byteBody = if (method != Method.HEAD) {
+      customEncodingHandler.orElse(PartialFunction.fromFunction(standardEncoding.tupled))(res.body(), encoding)
     } else {
       res.body()
     }
     val body = responseHandler(byteBody).handle(responseAs, responseMonad, responseMetadata)
     responseMonad.map(body)(Response(_, code, "", headers, Nil))
   }
 
+  private def standardEncoding: (InputStream, Option[String]) => InputStream = {
```

Comment:
shouldn't this be `(InputStream, String) => InputStream`? That is, the EncodingHandler would be only to specify behavior when there *is* any encoding?

No-encoding would be the default case.

Reply:
Right, fixed.